src/core/grid.js

```diff
@@ -156,6 +158,52 @@ class Grid {
       this.markAsMoved()
     }
   }
+
+  checkReached2048(value) {
+    if (value === 2048) this.reached2048 = true;
+  }
+
+  canMove() {
+    for (let row = 0; row < this.size; row++) {
+      for (let col = 0; col < this.size; col++) {
+        const currentCell = this.cells[row][col];
+
+        if (currentCell.isEmpty()) return true;
+
+        if (this.canMergeRight(row, col, currentCell) ||
+          this.canMergeDown(row, col, currentCell)) {
+          return true;
+        }
```

Comment:
Save the result of `canMergeRight` and `canMergeDown` in variables. It makes the `if` easier to read.
```suggestion
        const canMergeRight = this.canMergeRight(row, col, currentCell);
        const canMergeDown = this.canMergeDown(row, col, currentCell);
        
        if (canMergeRight || canMergeDown) return true;
```